course4/homework-forward-list/forward_list.hh

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <cstddef>
+#include <iostream>
+#include <iterator>
+#include <memory>
+
+namespace course {
+
+template<typename T>
+struct forward_list_node
+{
+	forward_list_node(const T& value, std::unique_ptr<forward_list_node<T>> next)
+		: value(value), next(std::move(next)) { }
```

Comment:
You copy-copy construct ```value``` here, which is OK... unless it isn't!
Extra assignment: try to make the list work for non-copyable types, for instance ```std::unique_ptr```